boards/beagle/beagleconnect_freedom/board_antenna.c

```diff
@@ -1,116 +1,119 @@
  * switch.
  */
 
-#include <zephyr/kernel.h>
-#include <zephyr/device.h>
+#define DT_DRV_COMPAT skyworks_sky13317
+
 #include <zephyr/init.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pinctrl.h>
 
 #include <ti/drivers/rf/RF.h>
-#include <driverlib/gpio.h>
-#include <driverlib/ioc.h>
 #include <driverlib/rom.h>
+#include <driverlib/interrupt.h>
 
-/* DIOs for RF antenna paths */
-#define BOARD_RF_HIGH_PA   29		/* TODO: pull from DT */
-#define BOARD_RF_SUB1GHZ   30		/* TODO: pull from DT */
+/* custom pinctrl states for the antenna mux */
+#define PINCTRL_STATE_ANT_SUBG    1
+#define PINCTRL_STATE_ANT_SUBG_PA 2
 
-static void board_cc13xx_rf_callback(RF_Handle client, RF_GlobalEvent events,
-		void *arg);
+#define BOARD_ANT_GPIO_PA   0
+#define BOARD_ANT_GPIO_SUBG 1
 
+static int board_antenna_init(const struct device *dev);
+static void board_cc13xx_rf_callback(RF_Handle client, RF_GlobalEvent events, void *arg);
 
 const RFCC26XX_HWAttrsV2 RFCC26XX_hwAttrs = {
-	.hwiPriority        = INT_PRI_LEVEL7,
-	.swiPriority        = 0,
+	.hwiPriority = INT_PRI_LEVEL7,
+	.swiPriority = 0,
 	.xoscHfAlwaysNeeded = true,
 	/* RF driver callback for custom antenna switching */
 	.globalCallback = board_cc13xx_rf_callback,
 	/* Subscribe to events */
-	.globalEventMask = (RF_GlobalEventRadioSetup |
-			RF_GlobalEventRadioPowerDown),
+	.globalEventMask = (RF_GlobalEventRadioSetup | RF_GlobalEventRadioPowerDown),
 };
 
+PINCTRL_DT_INST_DEFINE(0);
+DEVICE_DT_INST_DEFINE(0, board_antenna_init, NULL, NULL, NULL, POST_KERNEL,
+		      CONFIG_BOARD_ANTENNA_INIT_PRIO, NULL);
+
+#define GPIO_DT_SPEC(n, p, i) GPIO_DT_SPEC_GET_BY_IDX(n, p, i),
+
+static const struct pinctrl_dev_config *ant_pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(0);
+static const struct gpio_dt_spec ant_gpios[] = {
+	DT_FOREACH_PROP_ELEM(DT_NODELABEL(antenna_mux0), gpios, GPIO_DT_SPEC)};
+
 /**
  * Antenna switch GPIO init routine.
  */
-static int board_antenna_init(void)
+int board_antenna_init(const struct device *dev)
 {
+	ARG_UNUSED(dev);
+	int i;
 
-	/* set all paths to low */
-	IOCPinTypeGpioOutput(BOARD_RF_HIGH_PA);
-	GPIO_setOutputEnableDio(BOARD_RF_HIGH_PA, GPIO_OUTPUT_DISABLE);
-	IOCPinTypeGpioOutput(BOARD_RF_SUB1GHZ);
-	GPIO_setOutputEnableDio(BOARD_RF_SUB1GHZ, GPIO_OUTPUT_DISABLE);
+	/* default pinctrl configuration: set all antenna mux control pins as GPIOs */
+	pinctrl_apply_state(ant_pcfg, PINCTRL_STATE_DEFAULT);
+	/* set all GPIOs to 0 (all RF paths disabled) */
+	for (i = 0; i < ARRAY_SIZE(ant_gpios); i++) {
+		gpio_pin_configure_dt(&ant_gpios[i], 0);
+	}
 	return 0;
 }
 
-SYS_INIT(board_antenna_init, POST_KERNEL, CONFIG_BOARD_ANTENNA_INIT_PRIO);
-
+/**
+ * Custom TI RFCC26XX callback for switching the on-board antenna mux on radio setup.
+ */
 void board_cc13xx_rf_callback(RF_Handle client, RF_GlobalEvent events, void *arg)
 {
-	bool    sub1GHz   = false;
+	bool sub1GHz = false;
```

Comment:
The formatting changes are adding a lot of noise to the diff.

Reply:
Again, not simple formatting changes but rather different files. I can break it into another commit if it really is too big of a problem.

Reply:
splitting into 2 commits would help me. Not sure if it matters to the other reviewers.